mini-frame.el

```diff
@@ -310,15 +312,24 @@ ALIST is passed to `window--display-buffer'."
       (ignore which-key-popup-type)
       (unwind-protect
           (mini-frame--display fn args)
-        (when (frame-live-p mini-frame-completions-frame)
-          (make-frame-invisible mini-frame-completions-frame))
-        (when (frame-live-p mini-frame-selected-frame)
-          (select-frame-set-input-focus mini-frame-selected-frame))
-        (when (frame-live-p mini-frame-frame)
-          (if (eq system-type 'windows-nt)
-              ;; FIXME sometime buffer is not visible on windows
-              (delete-frame mini-frame-frame)
-            (make-frame-invisible mini-frame-frame))))))))
+        (when (= (recursion-depth) 0)
+          (when (frame-live-p mini-frame-completions-frame)
+            (make-frame-invisible mini-frame-completions-frame)
+            (modify-frame-parameters mini-frame-completions-frame '((parent-frame))))
+          (if (frame-parameter mini-frame-frame 'parent-frame)
+              (select-frame-set-input-focus
+               (frame-parameter mini-frame-frame 'parent-frame))
+            ;; trying to find an alternate frame to return the focus
+            (when-let (fr (cl-find-if-not
+                           (lambda (fr)
+                             (or (equal fr mini-frame-completions-frame)
+                                 (equal fr mini-frame-frame)
+                                 (frame-parameter fr 'parent-frame)))
+                           (frame-list)))
+              (select-frame-set-input-focus fr)))
```

Comment:
Can `(other-frame)` be used here to simplify the code?

Reply:
Since we're going to detach `mini-frame-completions-frame` and `mini-frame-frame` to `nil` parent, `(other-frame)` may just return them back. So I need this to find the first frame that's not the mini frame.